### ISSUE TITLE
Create accessible warning state for CMS fields with character limits

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#10135]
## Description
Within the CMS on the edit screens, fields that have a character limit enabled have a gold indicator when the editor gets close to the limit, within 5 characters. This gold color is used both on the border of the field and in the character count below the field.  However, the gold (Hex e09600) does not pass color contrast requirements against the white background.

NOTE: The border only needs to have a ratio of 3:1 as this is a UI element, not text.  However the text should meet 7:1 contrast requirements as ideally we want to hit the AAA requirement in this case.

## Screenshot
![image](https://user-images.githubusercontent.com/106678594/183948762-12138ac5-c375-4de0-84f2-b910fb38825f.png)

## Accessibility Standard

WCAG version 2.0 AAA, [Criterion 1.4.6](https://www.w3.org/WAI/WCAG21/Understanding/contrast-enhanced.html)

## Acceptance Criteria
- [x] Determine if the border color and the text color should be the same color, keep in mind that finding a gold color to pass 7:1 ratio requirements will be difficult
- [x] Interactive Design to create and document field warning state in Design System
- [x] Implementation ticket is created

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by EWashb** [comment]
@BlakeOrgan do you think this will be completed by the end of the sprint?

**Comment by BlakeOrgan** [comment]
@EWashb Yeah, shouldn't be a problem!

**Comment by BlakeOrgan** [comment]
To follow the existing patterns of character counters and to ensure that the text will always meet the 7:1 AAA contrast requirements, the number for the character counter in a warning state should always remain `#212121`. This update is also reflected in my recommendations about the error state of the character counter text color in ticket #10865.

Since the border is a UI element and only needs to have a ratio of 3:1, I recommend changing the yellow border to `#C58702`

@laflannery has confirmed these edits.

Figma design can be referenced here:
https://www.figma.com/file/gCSCL6VSGZGmYCMT0bq1Uu/Character-Counters?node-id=4%3A977&t=srvqt5e29PlK5GAd-1

The implementation ticket can be referenced here: #12377
